_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1041,11 +1043,76 @@ private function _process_onboarding( $data ) {
 			}
 		}
 
+		// Add/update business address widget
+		if ( isset( $data['businessAddress'] ) ) {
+			//grab the first sidebar and proceed if it is present
+			$first_sidebar = self::get_first_sidebar();
+
+			if( $first_sidebar ) {
+				$title = wp_unslash( $data[ 'businessAddress' ][ 'name' ] );
+				$address = wp_unslash(
+					$data[ 'businessAddress' ][ 'city' ] . ' ' .
+					$data[ 'businessAddress' ][ 'state' ] . ' ' .
+					$data[ 'businessAddress' ][ 'street' ] . ' ' .
+					$data[ 'businessAddress' ][ 'zip' ]
+				);
+				$widget_options = array(
+					'title'   => $title,
+					'address' => $address,
+					'phone'   => '',
+					'hours'   => '',
+					'showmap' => false,
+					'email' => '',
+					'apikey' => '',
+				);
+				$position = '0';
+				$id_base = 'widget_contact_info';
+
+				$has_ba_widget = self::has_business_info_widget( $first_sidebar );
```

Comment:
@AnnaMag one thing to consider here: for the GET endpoint, we'll need to be able to:

1. Know whether business address has been set or not
2. Retrieve all those business address fields, in order to be able to return them in the endpoint.

Right now, with your PR, for 1, we can use `self::has_business_info_widget( $first_sidebar )`, but for 2. we'll need something else, for example storing `$data[ 'businessAddress' ]` in an option. If you decide to do that, make sure to also add the option to the list of known options, in order it to be cleaned up after uninstalling JP (as done here: https://github.com/Automattic/jetpack/pull/8495/files#diff-cb166c81692a76d1a92d53cb5e80d67cR537).

Reply:
As per Slack chat, let's handle it in the follow-up. As always 💯 point.